packages/cactus-plugin-ledger-connector-iroha/src/main/json/openapi.json

```diff
@@ -231,6 +236,82 @@
       "PrometheusExporterMetricsResponse": {
         "type": "string",
         "nullable": false
+      },
+      "IrohaSocketSessionEvent": {
+        "type": "string",
+        "enum": [
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.Subscribe",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.Next",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.Unsubscribe",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.Error",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.Complete",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.SendAsyncRequest",
+            "org.hyperledger.cactus.api.async.iroha.SocketSession.SendSyncRequest"
```

Comment:
@outSH The `SendAsyncRequest` and `SendSyncRequest` could be their own endpoints with their own enums defined here in the openapi.json file so that they both can also have their own event names for subscribing, erroring, completion, etc.
Mixing all of them together (watchblocks and send async/sync requests) like this would mean that they are all crammed in on the same logical channel and if a client starts monitoring the blocks but then also "sends a request" and finally gets an error they would not know if the "send request" thingy is having issues or the watchblocks functionality.

Reply:
@petermetz Agree. I've moved `SendAsyncRequest` and `SendSyncRequest` into new `IrohaSocketIOTransactV1`. The logic in code is still handled by common `IrohaSocketIOEndpoint`. Ideally I'd like to come up with solution to have more protocol-agnostic endpoints, but we have different priorities right now.

Reply:
@outSH 

> Agree. I've moved SendAsyncRequest and SendSyncRequest into new IrohaSocketIOTransactV1.

Awesome, thank you!

> The logic in code is still handled by common IrohaSocketIOEndpoint. Ideally I'd like to come up with solution to have more protocol-agnostic endpoints, but we have different priorities right now.

I can live with that for now because it's not getting exported (from what I could tell) to the public API surface of the package so if we postpone renaming/refactoring it to later then it won't have a breaking change attached to it.

I opened an issue for tracking it: https://github.com/hyperledger/cactus/issues/2122